ironfish/src/wallet/account.ts

```diff
@@ -298,6 +379,10 @@ export class Account {
     return await this.walletDb.loadTransaction(this, hash, tx)
   }
 
+  async hasTransaction(hash: Buffer, tx?: IDatabaseTransaction): Promise<boolean> {
+    return (await this.getTransaction(hash, tx)) !== undefined
```

Comment:
we don't need to load the value + do deserialization, so we might as well make a version of hasTransaction that only loads the key.

Reply:
Good point, this is lazy